### DISCUSSION TITLE
How to clone and make changes in ultralytics repo

### DISCUSSION THREAD

**Discussion by Atharva-Malode** [ultralytics/ultralytics#10378] [General]
I was trying to clone the repo and make some changes to contribute to " specific class object counter ", Although was unable to do so. can anybody explain how to test whether they are working or not. Little about me I am a student who is currently pursuing my Bachelors and a beginner to. this doesn't worked for me

 Git clone
    Clone the `ultralytics` repository if you are interested in contributing to the development or wish to experiment with the latest source code. After cloning, navigate into the directory and install the package in editable mode `-e` using pip.
    bash
     Clone the ultralytics repository
    git clone https://github.com/ultralytics/ultralytics

    # Navigate to the cloned directory
    cd ultralytics

    # Install the package in editable mode for development
    pip install -e .

**Comment by pderrenger** [comment]
Hello! 👋 It's great to hear about your interest in contributing, especially with a focus on a "specific class object counter". To get started with cloning the repository and testing your changes, it looks like you're on the right track. Here's a condensed guide to ensure everything works smoothly:

1. **Clone the Repository**: Use Git to clone the Ultralytics repo:
   ```
   git clone https://github.com/ultralytics/ultralytics.git
   ```
2. **Navigate to the Repository's Directory**:
   ```
   cd ultralytics
   ```
3. **Install Dependencies** (If you haven't done so, it's crucial):
   ```
   pip install -r requirements.txt
   ```
4. **Install the Package in Editable Mode**:
    This step allows you to make live changes to the code and have them reflected without reinstallation.
   ```
   pip install -e .
   ```

To test your changes, consider writing small tests or use the existing ones by running them directly. You can also manually test by running specific functionalities or scripts that utilize the part of the code you've changed.

Since you're contributing to a "specific class object counter", after making your modifications, try running a detection with the model and verify if your counter logic is working as expected. Simple print statements or logging can help you see what's happening and confirm the efficacy of your updates.

Should you have any specific questions or run into problems, don't hesitate to ask! The community and maintainers are here to help. 😊

And remember, as a student and beginner, the learning journey can be challenging, but also rewarding. Keep experimenting, asking questions, and sharing your contributions. Your work could greatly benefit others!
